builtin/game/chat.lua

```diff
@@ -524,7 +524,7 @@ end
 
 -- Teleports player <name> to <p> if possible
 local function teleport_to_pos(name, p)
-	local lm = 31007 -- equals MAX_MAP_GENERATION_LIMIT in C++
+	local lm = tonumber(core.settings:get("mapgen_limit")) -- equals MAX_MAP_GENERATION_LIMIT in C++
```

Comment:
This change is simply wrong. This is how `MAX_MAP_GENERATION_LIMIT` is defined on the C++ side of things:

```lua
#define MAX_MAP_GENERATION_LIMIT (31007)
```

The Lua code clearly was the equivalent of that (although it should probably have gone in a global variable `minetest.MAX_MAP_GENERATION_LIMIT`. You're now making this setting-dependant. I also think this is intended to limit to the absolute world bounds and not the mapgen bounds, which may be reduced by admins who still want stuff like parkour arenas to work outside the specified bounds (which in fact may use the bounds to guarantee that no mapgen happens there). There's no mention of this in the description of this PR.

This is an unacceptable change in behavior IMO.